src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -48,6 +152,64 @@ struct boss_kazzara_the_hellforged : public BossAI
         }
     }
 
+    void JustDied(Unit* /*killer*/) override
+    {
+        Talk(SAY_DIED);
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+
+        me->RemoveAurasDueToSpell(SPELL_OVERRIDE_POWER_DISPLAY);
+        me->RemoveAurasDueToSpell(SPELL_PERIODIC_ENERGIZE);
+
+        summons.DespawnAll();
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+        Initialize();
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        Talk(SAY_AGGRO);
+        DoCastSelf(SPELL_OVERRIDE_POWER_DISPLAY, true);
+        DoCastSelf(SPELL_PERIODIC_ENERGIZE);
+        events.ScheduleEvent(EVENT_RAYS_OF_ANGUISH, 24s);
+        events.ScheduleEvent(EVENT_TERROR_CLAWS, 5s);
+        events.ScheduleEvent(EVENT_WINGS_OF_EXTINCTION, 14s);
+        events.ScheduleEvent(EVENT_DREAD_RIFT, 12s);
+        events.ScheduleEvent(EVENT_CHECK_ENERGY, 500ms);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 1);
+    }
+
+    void DamageTaken(Unit* /*killer*/, uint32& /*damage*/, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(SPELL_HELLSTEEL_CARNAGE_40_PCT, DIFFICULTY_NONE);
+
+        if (!spellInfo)
+            return;
+
+        if (me->HealthBelowPct(spellInfo->GetEffect(EFFECT_2).CalcValue(me)) && !_under80Percent)
+        {
+            Talk(SAY_ARMOR_80_PCT);
+            DoAction(ACTION_HEALTH_BELOW_80_PCT);
+        }
+        else if (me->HealthBelowPct(spellInfo->GetEffect(EFFECT_3).CalcValue(me)) && !_under60Percent)
```

Comment:
You will need update _under%Percent variables in order to don't trigger this more of 1 time.